### PR TITLE
[native] Update build to support new dependency versions

### PR DESCRIPTION
**Pull request by kgpai**
## Description
FMT is updated to 10.1.1.
Folly, Fizz, Wangle, FBThrift are updated to v2023.12.04.00.
New dependency mvfst v2023.12.04.00 added.
The specific requirement of OpenSSL@1.1 is now removed.

Users will have to reinstall the dependencies

## Motivation and Context
This change should allow you to build Prestissimo with versions of fmt above 9.  Velox updated its dependency to fmt 10. Users who update to fmt 10 will not be able to build presto_native. This change fixes presto_native so that it can build with fmt 9 and above.

## Impact
Users who are on fmt 8 will need to update their systems by running `./scripts/setup-<os>.sh`.

## Test Plan
Tested locally

```
== NO RELEASE NOTE ==
```

Resolves https://github.com/prestodb/presto/issues/21792

### CODE REVIEW

**Comment by majetideepak** [comment]
@kgpai can you use this image majetideepak4/prestocpp:root-20240126 to verify? We can then upload it to the prestodb account.

**Comment by majetideepak** [comment]
@kgpai here is the CI image managed by the Presto release team public.ecr.aws/oss-presto/prestocpp:root-20240126
Can you please use this in the PR? Thanks.

**Review by majetideepak — APPROVED** [review]

**Comment by majetideepak** [comment]
@tdcmeehan, @mbasmanova, can one you please approve from the Committers group? Thanks.

**Review by tdcmeehan — APPROVED** [review]

**Comment by majetideepak** [comment]
@kgpai can you fix setup-ubuntu and setup-centos as well?

File: presto-native-execution/scripts/setup-centos.sh
Comment (majetideepak) [inline presto-native-execution/scripts/setup-centos.sh:66]: We don't need libsodium since we install the package `libsodium-devel` in the Velox script.

Comment (kgpai) [reply presto-native-execution/scripts/setup-centos.sh:66]: Good point will remove.

**Review by majetideepak — APPROVED** [review]

**Comment by majetideepak** [comment]
@tdcmeehan, @mbasmanova can you please approve this once again? This should fix the MacOS build. Thanks!

File: .circleci/continue_config.yml
Comment (mbasmanova) [inline .circleci/continue_config.yml:63]: Do we have some documentation somewhere that explains where these images come from and how to update them? Would it be possible to add a comment with a link here?

File: presto-native-execution/presto_cpp/main/tests/PrestoQueryRunner.cpp
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/tests/PrestoQueryRunner.cpp:102]: We should delete this file. The code was moved to Velox.

File: presto-native-execution/scripts/setup-macos.sh
Comment (mbasmanova) [inline presto-native-execution/scripts/setup-macos.sh:62]: Why are we no longer installing fizz and others?

File: .circleci/continue_config.yml
Comment (majetideepak) [reply .circleci/continue_config.yml:63]: We have instructions in the README here https://github.com/prestodb/presto/tree/master/presto-native-execution#build-using-dockerfile
The update will be handled by the Presto release management team. https://github.com/orgs/prestodb/teams/team-releases

File: presto-native-execution/scripts/setup-macos.sh
Comment (majetideepak) [reply presto-native-execution/scripts/setup-macos.sh:62]: Fizz and others have been moved to the Velox setup scripts.

Comment (mbasmanova) [reply presto-native-execution/scripts/setup-macos.sh:62]: > Fizz and others have been moved to the Velox setup scripts.

I missed that. That explains why we started seeing fizz-related failures on PRs. Looks like we didn't communicate addition of these dependencies to Velox.

https://github.com/facebookincubator/velox/pull/8494?fbclid=IwAR19JnG6_rot86pZzqaZWxF-g60zP7juSEH7gwoW01n6vMqM6r9Y3-rl68g#issuecomment-1916815212

File: presto-native-execution/presto_cpp/main/tests/PrestoQueryRunner.cpp
Comment (majetideepak) [reply presto-native-execution/presto_cpp/main/tests/PrestoQueryRunner.cpp:102]: Can we do this in a follow-up PR?

File: presto-native-execution/scripts/setup-macos.sh
Comment (majetideepak) [reply presto-native-execution/scripts/setup-macos.sh:62]: This is the PR that updated the Velox side scripts https://github.com/facebookincubator/velox/pull/8454

Comment (majetideepak) [reply presto-native-execution/scripts/setup-macos.sh:62]: > That explains why we started seeing fizz-related failures on PRs. Looks like we didn't communicate addition of these dependencies to Velox.

This was not new for Velox. It was added here originally https://github.com/facebookincubator/velox/pull/5743/files
The failure could be due to the newer versions. I will take a look. Did not notice that issue.

File: presto-native-execution/presto_cpp/main/tests/PrestoQueryRunner.cpp
Comment (majetideepak) [reply presto-native-execution/presto_cpp/main/tests/PrestoQueryRunner.cpp:102]: PR here https://github.com/prestodb/presto/pull/21822

**Comment by majetideepak** [comment]
@mbasmanova I addressed your comments. Can you take another look? Thanks!

File: presto-native-execution/scripts/setup-macos.sh
Comment (mbasmanova) [inline presto-native-execution/scripts/setup-macos.sh:20]: Are we hard-coding this in 2 places: Velox and here? Is this how we want this to be? Why can't we use the version specified in Velox?

Comment (mbasmanova) [reply presto-native-execution/scripts/setup-macos.sh:20]: Strangely I don't see similar change in other setup-xxx.sh scripts. Why are they different?

File: presto-native-execution/presto_cpp/main/operators/tests/UnsafeRowShuffleTest.cpp
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/operators/tests/UnsafeRowShuffleTest.cpp:1180]: It would be nice to extract these changes into a separate PR. I believe they do not require the upgrades. This would help keep this PR small and focused.

File: presto-native-execution/presto_cpp/main/PeriodicTaskManager.cpp
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/PeriodicTaskManager.cpp:711]: Why do we need this? Shouldn't this type of change go into wherever MemoryArbitrator::Stats are defined? If Velox is later updated to add a similar change, will this cause a conflict?

File: presto-native-execution/presto_cpp/main/operators/tests/UnsafeRowShuffleTest.cpp
Comment (majetideepak) [reply presto-native-execution/presto_cpp/main/operators/tests/UnsafeRowShuffleTest.cpp:1180]: These are related to the folly upgrade that is part of the Velox upgrade here.

File: presto-native-execution/scripts/setup-macos.sh
Comment (majetideepak) [reply presto-native-execution/scripts/setup-macos.sh:20]: I believe Velox script gets invoked in a subshell whose environment might not be visible to the parent script here. So we need to define here as well.

Comment (majetideepak) [reply presto-native-execution/scripts/setup-macos.sh:20]: I will take a look if we can cache the Velox variable. But out of scope here.

File: presto-native-execution/presto_cpp/main/PeriodicTaskManager.cpp
Comment (majetideepak) [reply presto-native-execution/presto_cpp/main/PeriodicTaskManager.cpp:711]: Good point! @kgpai do you know the answer to this?

Comment (majetideepak) [reply presto-native-execution/presto_cpp/main/PeriodicTaskManager.cpp:711]: I think these should live in Velox. Added a PR here https://github.com/facebookincubator/velox/pull/8608

**Review by tanjialiang — APPROVED** [review]

**Comment by tanjialiang** [comment]
Thanks, shall we land this to unblock?

**Pull request merged**
